src/common/metrics.rs

```diff
@@ -806,138 +806,245 @@ fn label_pair(name: &str, value: &str) -> LabelPair {
-            prefix,
-        ));
+    /// Returns the minor and major page faults of all children including grandchildren, etc.
+    pub fn faults_for_all_children(parent: Pid) -> Result<(u64, u64), ProcError> {
+        let children = child_processes_helper(parent)?;
+        let (mut min, mut maj) = (0, 0);
+        for child in children {
+            let stat = match Process::new(child).and_then(|i| i.stat()) {
+                Ok(proc) => proc,
+                // Ignore children that don't exist anymore.
+                Err(ProcError::NotFound(_)) => continue,
+                Err(err) => return Err(err),
+            };
 
-        metrics.push(metric_family(
-            "process_open_mmaps",
-            "count of open mmaps",
-            MetricType::GAUGE,
-            vec![gauge(self.mmap_count as f64, &[])],
-            prefix,
-        ));
+            min += stat.minflt + stat.cminflt;
+            maj += stat.majflt + stat.cmajflt;
+        }
+        Ok((min, maj))
+    }
 
-        metrics.push(metric_family(
-            "system_max_mmaps",
-            "system wide limit of open mmaps",
-            MetricType::GAUGE,
-            vec![gauge(self.system_mmap_limit as f64, &[])],
-            prefix,
-        ));
+    /// Returns a list of all children of a given process. This works recursively, including grandchildren, etc.
+    fn child_processes_helper(pid: Pid) -> Result<AHashSet<Pid>, ProcError> {
+        let mut pids = AHashSet::default();
+        child_processes_recursive(pid, &mut pids)?;
 
-        metrics.push(metric_family(
-            "process_open_fds",
-            "count of currently open file descriptors",
-            MetricType::GAUGE,
-            vec![gauge(self.open_fds as f64, &[])],
-            prefix,
-        ));
+        // Exclude parent Pid from results
+        pids.remove(&pid);
 
-        let fds_limit = match (self.max_fds_soft, self.max_fds_hard) {
-            (0, hard) => hard,                         // soft unlimited, use hard
-            (soft, 0) => soft,                         // hard unlimited, use soft
-            (soft, hard) => std::cmp::min(soft, hard), // both limited, use minimum
-        };
-        metrics.push(metric_family(
-            "process_max_fds",
-            "limit for open file descriptors",
-            MetricType::GAUGE,
-            vec![gauge(fds_limit as f64, &[])],
-            prefix,
-        ));
+        Ok(pids)
+    }
 
-        metrics.push(metric_family(
-            "process_minor_page_faults_total",
-            "count of minor page faults which didn't cause a disk access",
-            MetricType::COUNTER,
-            vec![counter(self.minor_page_faults as f64, &[])],
-            prefix,
-        ));
+    /// Recursively collects all children of a process, specified by `pid`.
+    fn child_processes_recursive(pid: Pid, pids: &mut AHashSet<Pid>) -> Result<(), ProcError> {
+        for child_pid in child_processes(pid)? {
```

Comment:
Very minor performance nit: `child_processes` allocates a vector that would be dropped right after iterating. Tho, I think it's far from being a bottleneck.